### ISSUE TITLE
AttributeError: 'Session' object has no attribute 'modified'

### ISSUE THREAD

**Issue by cjnolet** [miguelgrinberg/Flask-SocketIO#475]
I'm new to flask and very new to the flask socketio library so it's very likely I'm doing something wrong. I've had a flask webserver running for a few weeks no problem and I've recently tried adding the SocketIO support to it.  The normal http server endpoints appear to work fine but when trying to connect a socketio client, I'm getting the following exception:
```Connect handler raised an exception
Traceback (most recent call last):
  File "/home/socialmix/.local/lib/python2.7/site-packages/gevent/pywsgi.py", line 936, in handle_one_response
    self.run_application()
  File "/home/socialmix/.local/lib/python2.7/site-packages/gevent/pywsgi.py", line 909, in run_application
    self.result = self.application(self.environ, self.start_response)
  File "/usr/lib64/python2.7/site-packages/flask/app.py", line 1997, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/lib64/python2.7/site-packages/beaker/middleware.py", line 156, in __call__
    return self.wrap_app(environ, session_start_response)
  File "/home/socialmix/.local/lib/python2.7/site-packages/flask_socketio/__init__.py", line 42, in __call__
    start_response)
  File "/home/socialmix/.local/lib/python2.7/site-packages/engineio/middleware.py", line 47, in __call__
    return self.engineio_app.handle_request(environ, start_response)
  File "/home/socialmix/.local/lib/python2.7/site-packages/socketio/server.py", line 353, in handle_request
    return self.eio.handle_request(environ, start_response)
  File "/home/socialmix/.local/lib/python2.7/site-packages/engineio/server.py", line 252, in handle_request
    transport, b64)
  File "/home/socialmix/.local/lib/python2.7/site-packages/engineio/server.py", line 370, in _handle_connect
    raise reraise_exc
AttributeError: 'Session' object has no attribute 'modified'
Wed May 31 14:06:23 2017 {'REMOTE_PORT': '56286', 'HTTP_HOST': 'localhost:5000', 'REMOTE_ADDR': '127.0.0.1', (hidden keys: 27)} failed with AttributeError
```
Here's how I'm creating and running the webserver app:

```

session_opts = {
    'session.type': 'file',
    'session.data_dir':'/tmp',
    'session.auto': True
}



class BeakerSessionInterface(SessionInterface):
    def open_session(self, app, request):
        
        session = request.environ['beaker.session']
        return session

    def save_session(self, app, session, response):
        session.save()

socketio = SocketIO(app)
app.wsgi_app = SessionMiddleware(app.wsgi_app, session_opts)
app.session_interface = BeakerSessionInterface()
```

**Comment by miguelgrinberg** [comment]
Hmm. The problem is that this pattern that you are using to implement beaker sessions exposes a session class that is not compatible with Flask's own. In particular, on a Flask `Session` object you can say `session.modified` and that tells you if there has been any changes to the session within the request. The `modified` attribute does not exist in the beaker session object.

So this is really not your fault, this project makes the assumption that you will be using a session object that is similar to Flask's. For example, the server-side sessions implemented by the [Flask-Sessions](https://github.com/fengsp/flask-session) extension are compatible with Flask, as they inherit from Flask's `SessionMixin` class.

So I need to think about how to support beaker sessions. Maybe I need to assume that the session was modified if I can't ask the session itself.

**Comment by cjnolet** [comment]
Miguel,

Is there a different pattern I could use just to make it work? I don't mind
changing the way I'm using sessions. I thought the SessionInterface
middleware was neat for automating the pulling of the session out of the
request but if it's causing a problem with the socketio stuff, I can do
away with it.

**Comment by miguelgrinberg** [comment]
If you switch to the default Flask sessions this problem will go away. At least you can do that until I figure out how to support both Flask and Beaker sessions together.

**Comment by miguelgrinberg** [comment]
@cjnolet Can I ask you to install the master branch directly out of this repository and see if your beaker sessions work well? To install from github you can use the following command:

    pip install https://github.com/miguelgrinberg/Flask-SocketIO/archive/master.zip

**Comment by miguelgrinberg** [comment]
This should now be fixed in the 2.9.0 release of this package.
